trieste/acquisition/function.py

```diff
@@ -56,7 +56,25 @@
 """
 
 
-class AcquisitionFunctionBuilder(ABC):
+T = TypeVar("T")
+""" Unbound type variable. """
+
+
+class Empiric(ABC, Generic[T]):
+    """An :class:`Empiric` constructs a value of some type `T` from the current data and models."""
+
+    @abstractmethod
+    def acquire(
+        self, datasets: Mapping[str, Dataset], models: Mapping[str, ProbabilisticModel]
+    ) -> T:
+        """
+        :param datasets: The data from the observer.
+        :param models: The models over each dataset in ``datasets``.
+        :return: A value of type `T`.
+        """
+
+
+class AcquisitionFunctionBuilder(Empiric[AcquisitionFunction], ABC):
     """An :class:`AcquisitionFunctionBuilder` builds an acquisition function."""
```

Comment:
Whats going on here?

Reply:
i need to discuss this with you. I've abstracted the notion of an acquisition builder, and `prepare_acquisition_function` isn't an appropriate name when we're not building an acquisition function. I didn't want to change `prepare_acquisition_function` everywhere until I'd got your opinion on it

Reply:
really, I envisage `Empiric` being a type alias so there's nothing to name (I didn't do that here cos it's out of scope), but we'll need a name at least for now

Reply:
i realise I don't need to do this in this PR. I do recommend it, but it's not necessary here